Cython/Compiler/PyrexTypes.py

```diff
@@ -3807,9 +3807,8 @@ def create_from_py_utility_code(self, env):
                 'type': self.cname,
             })
             # Override directives that should not be inherited from user code.
-            # TODO: filter directives with an allow list to keep only those that are safe and relevant.
-            directives = dict(env.directives, cpp_locals=False)
             from .UtilityCode import CythonUtilityCode
+            directives = CythonUtilityCode.filter_directives(env.directives)
```

Comment:
The `Override directives …` comment seems ambiguous (override user directives or defaults?), but otherwise helpful, which suggests that the method names isn't sufficiently concrete. `filter` is a fairly broad terms that doesn't hint at why or how. `filter_inherited_directives()`, maybe? Can't think of something better right now.